clippy_lints/src/methods/unnecessary_fallible_conversions.rs

```diff
@@ -45,47 +46,80 @@ fn check<'tcx>(
+        //
+        // `try_into().unwrap()` specifically can be trivially replaced with just `into()`,
+        // so that can be machine-applicable
         let parent_unwrap_call = get_parent_expr(cx, expr).and_then(|parent| {
             if let ExprKind::MethodCall(path, .., span) = parent.kind
                 && let sym::unwrap | sym::expect = path.ident.name
             {
-                Some(span)
+                // include `.` before `unwrap`/`expect`
+                Some(span.with_lo(expr.span.hi()))
             } else {
                 None
             }
         });
-        let (source_ty, target_ty, sugg, span, applicability) = match kind {
-            FunctionKind::TryIntoMethod if let Some(unwrap_span) = parent_unwrap_call => {
-                // Extend the span to include the unwrap/expect call:
-                // `foo.try_into().expect("..")`
-                //      ^^^^^^^^^^^^^^^^^^^^^^^
-                //
-                // `try_into().unwrap()` specifically can be trivially replaced with just `into()`,
-                // so that can be machine-applicable
 
-                (
-                    self_ty,
-                    other_ty,
-                    "into()",
-                    primary_span.with_hi(unwrap_span.hi()),
-                    Applicability::MachineApplicable,
-                )
+        // If there is an unwrap/expect call, extend the span to include the call
+        let span = if let Some(unwrap_call) = parent_unwrap_call {
+            primary_span.with_hi(unwrap_call.hi())
+        } else {
+            primary_span
+        };
+
+        let qpath_spans = qpath.and_then(|qpath| match qpath {
+            QPath::Resolved(_, path) => {
+                let segments = path.segments.iter().map(|seg| seg.ident).collect::<Vec<_>>();
+                (segments.len() == 2).then(|| vec![segments[0].span, segments[1].span])
+            },
+            QPath::TypeRelative(_, seg) => Some(vec![seg.ident.span]),
+            QPath::LangItem(_, _) => unreachable!("`TryFrom` and `TryInto` are not lang items"),
+        });
+
+        let (source_ty, target_ty, sugg, applicability) = match (kind, &qpath_spans, parent_unwrap_call) {
+            (FunctionKind::TryIntoMethod, _, Some(unwrap_span)) => {
+                let sugg = vec![(primary_span, String::from("into")), (unwrap_span, String::new())];
+                (self_ty, other_ty, sugg, Applicability::MachineApplicable)
+            },
+            (FunctionKind::TryFromFunction, Some(spans), Some(unwrap_span)) => {
+                let sugg = match spans.len() {
+                    1 => vec![(spans[0], String::from("from")), (unwrap_span, String::new())],
+                    2 => vec![
+                        (spans[0], String::from("From")),
+                        (spans[1], String::from("from")),
+                        (unwrap_span, String::new()),
+                    ],
+                    _ => unreachable!(),
```

Comment:
Can we use destructuring here? As an example:
```suggestion
                let sugg = match &*spans {
                    [first] => vec![(first, String::from("from")), (unwrap_span, String::new())],
                    [first, second] => vec![
                        (first, String::from("From")),
                        (second, String::from("from")),
                        (unwrap_span, String::new()),
                    ],
                    _ => unreachable!(),
```
If you want you could also use an enum to remove the wildcard branch, like: `OneSpan(_)` + `TwoSpans(_, _)`.